src/soap_wsse/signing.py

```diff
@@ -131,14 +135,18 @@ def verify_envelope(reply, key_file):
     dsigCtx = xmlsec.DSigCtx()
 
     xmlsec.addIDs(doc, ['Id'])
-    signKey = xmlsec.Key.load(key_file, xmlsec.KeyDataFormatPem)
+    signKey = xmlsec.Key.load(
+        key_file,
+        xmlsec.KeyDataFormatPem,
+        password,
+    )
     signKey.name = os.path.basename(key_file)
 
     dsigCtx.signKey = signKey
     try:
         dsigCtx.verify(node)
-    except xmlsec.VerificationError:
-        return False
+    except xmlsec.VerificationError, e:
+        return e.args[1] == 2
```

Comment:
Also `e.args[1] == 2` will actually return True if the exception is `('signature verification failed', 2)` while this should return False here i suppose?